Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/ui/settings/SettingsFragmentPresenter.java

```diff
@@ -291,13 +313,395 @@ private void addHackSettings(ArrayList<SettingsItem> sl)
+
+			sl.add(new HeaderSetting(null, null, R.string.controller_c, 0));
+			sl.add(new InputBindingSetting(SettingsFile.KEY_GCBIND_C_UP + gcPadNumber, SettingsFile.SECTION_BINDINGS, R.string.generic_up, bindCUp));
+			sl.add(new InputBindingSetting(SettingsFile.KEY_GCBIND_C_DOWN + gcPadNumber, SettingsFile.SECTION_BINDINGS, R.string.generic_down, bindCDown));
+			sl.add(new InputBindingSetting(SettingsFile.KEY_GCBIND_C_LEFT + gcPadNumber, SettingsFile.SECTION_BINDINGS, R.string.generic_left, bindCLeft));
+			sl.add(new InputBindingSetting(SettingsFile.KEY_GCBIND_C_RIGHT + gcPadNumber, SettingsFile.SECTION_BINDINGS, R.string.generic_right, bindCRight));
+
+			sl.add(new HeaderSetting(null, null, R.string.controller_trig, 0));
+			sl.add(new InputBindingSetting(SettingsFile.KEY_GCBIND_TRIGGER_L + gcPadNumber, SettingsFile.SECTION_BINDINGS, R.string.trigger_left, bindTriggerL));
+			sl.add(new InputBindingSetting(SettingsFile.KEY_GCBIND_TRIGGER_R + gcPadNumber, SettingsFile.SECTION_BINDINGS, R.string.trigger_right, bindTriggerR));
+
+			sl.add(new HeaderSetting(null, null, R.string.controller_dpad, 0));
+			sl.add(new InputBindingSetting(SettingsFile.KEY_GCBIND_DPAD_UP + gcPadNumber, SettingsFile.SECTION_BINDINGS, R.string.generic_up, bindDPadUp));
+			sl.add(new InputBindingSetting(SettingsFile.KEY_GCBIND_DPAD_DOWN + gcPadNumber, SettingsFile.SECTION_BINDINGS, R.string.generic_down, bindDPadDown));
+			sl.add(new InputBindingSetting(SettingsFile.KEY_GCBIND_DPAD_LEFT + gcPadNumber, SettingsFile.SECTION_BINDINGS, R.string.generic_left, bindDPadLeft));
+			sl.add(new InputBindingSetting(SettingsFile.KEY_GCBIND_DPAD_RIGHT + gcPadNumber, SettingsFile.SECTION_BINDINGS, R.string.generic_right, bindDPadRight));
+		}
+		else // Adapter
+		{
+			Setting rumble = mSettings.get(SettingsFile.SECTION_CORE).getSetting(SettingsFile.KEY_GCADAPTER_RUMBLE + gcPadNumber);
+			Setting bongos = mSettings.get(SettingsFile.SECTION_CORE).getSetting(SettingsFile.KEY_GCADAPTER_BONGOS + gcPadNumber);
+
+			sl.add(new CheckBoxSetting(SettingsFile.KEY_GCADAPTER_RUMBLE + gcPadNumber, SettingsFile.SECTION_CORE, R.string.gc_adapter_rumble, R.string.gc_adapter_rumble_description, false, rumble));
+			sl.add(new CheckBoxSetting(SettingsFile.KEY_GCADAPTER_BONGOS + gcPadNumber, SettingsFile.SECTION_CORE, R.string.gc_adapter_bongos, R.string.gc_adapter_bongos_description, false, bongos));
+		}
+	}
+
+	private void addWiimoteSubSettings(ArrayList<SettingsItem> sl, int wiimoteNumber)
 	{
-		Setting rumble = mSettings.get(SettingsFile.SECTION_CORE).getSetting(SettingsFile.KEY_GCADAPTER_RUMBLE + gcPadNumber);
-		Setting bongos = mSettings.get(SettingsFile.SECTION_CORE).getSetting(SettingsFile.KEY_GCADAPTER_BONGOS + gcPadNumber);
+		Setting extension = mSettings.get(SettingsFile.SECTION_BINDINGS).getSetting(SettingsFile.KEY_WIIMOTE_EXTENSION + (wiimoteNumber - 3));
```

Comment:
The Wii binding settings continue off the GameCube settings for controller number. GC controllers are 0-3, Wii controllers are 4-7, so WiimoteA_4 in the main ini is the mapping for Wiimote 1 button A. In the Wii ini, where we set the extension, the controllers are 1-4.